File: src/server/src/main/java/com/project/infrastructure/config/CloudinaryConfig.java
package com.project.infrastructure.config;

import com.cloudinary.Cloudinary;
import io.github.cdimascio.dotenv.Dotenv;
import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Factory;

@Factory
public class CloudinaryConfig {

    @Bean
    public Cloudinary cloudinary() {
        var dotEnv = Dotenv.configure().directory("misc/.env").load();
        var cloudinary = new Cloudinary(dotEnv.get("CLOUDINARY_URL"));
        cloudinary.config.secure = true;

        return cloudinary;
    }
}

File: src/server/src/main/java/com/project/infrastructure/config/CloudinaryConfig.kt
package com.project.infrastructure.config

import com.cloudinary.Cloudinary
import io.github.cdimascio.dotenv.Dotenv
import io.micronaut.context.annotation.Bean
import io.micronaut.context.annotation.Factory

@Factory
class CloudinaryConfig {
    @Bean
    fun cloudinary(): Cloudinary {
        val dotEnv = Dotenv.configure().directory("./misc").filename(".env").load()
        val cloudinary = Cloudinary(dotEnv["CLOUDINARY_URL"])
        cloudinary.config.secure = true

        return cloudinary
    }
}
